File: agent/commands-generator-api/src/main/java/org/matilda/commands/types/ByteStringConverter.java
package org.matilda.commands.types;

import com.google.protobuf.ByteString;
import com.google.protobuf.BytesValue;

public class ByteStringConverter implements ProtobufConverter<ByteString> {
    @Override
    public BytesValue convert(ByteString object) {
        return BytesValue.newBuilder().setValue(object).build();
    }
}

File: agent/commands-generator-api/src/main/java/org/matilda/commands/types/ByteStringConverter.kt
package org.matilda.commands.types

import com.google.protobuf.Any
import com.google.protobuf.ByteString
import com.google.protobuf.BytesValue

class ByteStringConverter : ProtobufConverter<ByteString> {
    override fun convertToProtobuf(obj: ByteString): BytesValue {
        return BytesValue.newBuilder().setValue(obj).build()
    }

    override fun convertFromProtobuf(obj: Any): ByteString {
        return obj.unpack(BytesValue::class.java).value
    }
}
